<launch>
    <arg name="DEBUG_OUTPUT" default="false"/>
    <arg name="config" default="$(find ros_ndi_tracker)/config/polaris_config.yaml"/>
    <arg name="config_param_name" default="polaris_sensor/marker"/>
    <arg name="tracker_type" default="polaris"/>
    <arg name="port" default="/dev/ttyUSB2"/>
    <arg name="base_link_name" default="polaris_base_link"/>
    <arg name="use_rivz" default="false"/>
    
    <node name="polaris_sensor" pkg="ros_ndi_tracker" type="broadcast_ndi_sensor.py" output="screen">
        <rosparam file="$(arg config)" command="load" subst_value="True"/>
        <param name="config_param_name" value="$(arg config_param_name)"/>
        <param name="base_link_name" value="$(arg base_link_name)"/>
        <param name="port" value="$(arg port)"/>
        <param name="tracker_type" value="$(arg tracker_type)"/>
        <param name="DEBUG_OUTPUT" value="$(arg DEBUG_OUTPUT)"/>
    </node>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find ros_ndi_tracker)/config/sample.rviz" if="$(arg use_rivz)" /> 

</launch>